utils/python/CIME/case_test.py

```diff
@@ -6,7 +6,30 @@
 from CIME.utils import expect, find_system_test, append_status
 from CIME.SystemTests.system_tests_common import *
 
-import sys
+import sys, signal
+
+def _iter_signal_names():
+    for signame in [item for item in dir(signal) if item.startswith("SIG") and not item.startswith("SIG_")]:
+        yield signame
+
+def _signal_handler(signum, _):
+    name = "Unknown"
+    for signame in _iter_signal_names():
+        if signum == getattr(signal, signame):
+            name = signame
+
+    expect(False, "Job killed due to receiving signal %d (%s)" % (signum, name))
+
+def _set_up_signal_handlers():
+    """
+    Add handles for all signals that might be used to abort a test
+
+    We need to handle a wide variety due to different implementations of the
+    timeout mechanism for different batch systems.
+    """
+    for signame in ["SIGINT", "SIGTERM", "SIGXCPU", "SIGUSR1", "SIGUSR2"]:
```

Comment:
Not a bad idea longerm. I think for now this list of signals is safe to handle on any batch system.